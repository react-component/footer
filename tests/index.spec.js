import React from 'react';
import { mount } from 'enzyme';
import Footer from '../index';

describe('rc-footer', () => {
  it('render empty Footer', () => {
    const wrapper = mount(<Footer />);
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('render Footer', () => {
    const wrapper = mount(
      <Footer
        columns={[
          {
            title: '相关资源',
            items: [
              {
                title: 'Ant Design Pro',
                url: 'https://pro.ant.design/',
                openExternal: true,
              },
              {
                title: 'Ant Design Mobile',
                url: 'https://mobile.ant.design/',
                openExternal: true,
                className: 'my-class-name',
                style: {
                  color: 'blue',
                },
              },
              {
                title: 'Kitchen',
                url: 'https://kitchen.alipay.com/',
                description: 'Sketch 工具集',
                LinkComponent: 'span',
              },
            ],
          },
          {
            title: '社区',
            className: 'my-class-name',
            style: {
              color: 'red',
            },
            items: [
              {
                title: 'Ant Design Pro',
                url: 'https://pro.ant.design/',
                openExternal: true,
              },
              {
                title: 'Ant Design Mobile',
                url: 'https://mobile.ant.design/',
                openExternal: true,
              },
              {
                title: 'Kitchen',
                url: 'https://kitchen.alipay.com/',
                description: 'Sketch 工具集',
              },
            ],
          },
          {
            title: '帮助',
            items: [
              {
                title: 'Ant Design Pro',
                url: 'https://pro.ant.design/',
                openExternal: true,
              },
              {
                title: 'Ant Design Mobile',
                url: 'https://mobile.ant.design/',
                openExternal: true,
              },
              {
                title: 'Kitchen',
                url: 'https://kitchen.alipay.com/',
                description: 'Sketch 工具集',
              },
            ],
          },
          {
            icon: (
              <img
                src="https://gw.alipayobjects.com/zos/rmsportal/nBVXkrFdWHxbZlmMbsaH.svg"
                alt="more products"
              />
            ),
            title: '更多产品',
            items: [
              {
                icon: (
                  <img
                    src="https://gw.alipayobjects.com/zos/rmsportal/XuVpGqBFxXplzvLjJBZB.svg"
                    alt="yuque"
                  />
                ),
                title: '语雀',
                url: 'https://yuque.com',
                description: '知识创作与分享工具',
                openExternal: true,
              },
              {
                icon: (
                  <img
                    src="https://gw.alipayobjects.com/zos/rmsportal/uHocHZfNWZOdsRUonZNr.png"
                    alt="yuque"
                  />
                ),
                title: '云凤蝶',
                url: 'https://yunfengdie.com',
                description: '中台建站平台',
                openExternal: true,
              },
            ],
          },
        ]}
        bottom="Made with ❤️ by AFX"
      />,
    );
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('render Footer with columnLayout and backgroundColor', () => {
    const wrapper = mount(
      <Footer
        columns={[{ title: 'title' }, { title: 'title' }]}
        columnLayout="space-between"
        backgroundColor="transparent"
        style={{ color: 'red' }}
      />,
    );
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('render light theme Footer', () => {
    const wrapper = mount(
      <Footer
        columns={[{ title: 'title' }, { title: 'title' }]}
        theme="light"
        style={{ color: 'red' }}
      />,
    );
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('render Footer with maxColumnsPerRow', () => {
    const wrapper = mount(
      <Footer
        maxColumnsPerRow={2}
        columns={[
          {
            title: '相关资源',
            items: [
              {
                title: 'Ant Design Pro',
                url: 'https://pro.ant.design/',
                openExternal: true,
              },
              {
                title: 'Ant Design Mobile',
                url: 'https://mobile.ant.design/',
                openExternal: true,
                className: 'my-class-name',
                style: {
                  color: 'blue',
                },
              },
              {
                title: 'Kitchen',
                url: 'https://kitchen.alipay.com/',
                description: 'Sketch 工具集',
                LinkComponent: 'span',
              },
            ],
          },
          {
            title: '社区',
            className: 'my-class-name',
            style: {
              color: 'red',
            },
            items: [
              {
                title: 'Ant Design Pro',
                url: 'https://pro.ant.design/',
                openExternal: true,
              },
              {
                title: 'Ant Design Mobile',
                url: 'https://mobile.ant.design/',
                openExternal: true,
              },
              {
                title: 'Kitchen',
                url: 'https://kitchen.alipay.com/',
                description: 'Sketch 工具集',
              },
            ],
          },
          {
            title: '帮助',
            items: [
              {
                title: 'Ant Design Pro',
                url: 'https://pro.ant.design/',
                openExternal: true,
              },
              {
                title: 'Ant Design Mobile',
                url: 'https://mobile.ant.design/',
                openExternal: true,
              },
              {
                title: 'Kitchen',
                url: 'https://kitchen.alipay.com/',
                description: 'Sketch 工具集',
              },
            ],
          },
          {
            icon: (
              <img
                src="https://gw.alipayobjects.com/zos/rmsportal/nBVXkrFdWHxbZlmMbsaH.svg"
                alt="more products"
              />
            ),
            title: '更多产品',
            items: [
              {
                icon: (
                  <img
                    src="https://gw.alipayobjects.com/zos/rmsportal/XuVpGqBFxXplzvLjJBZB.svg"
                    alt="yuque"
                  />
                ),
                title: '语雀',
                url: 'https://yuque.com',
                description: '知识创作与分享工具',
                openExternal: true,
              },
              {
                icon: (
                  <img
                    src="https://gw.alipayobjects.com/zos/rmsportal/uHocHZfNWZOdsRUonZNr.png"
                    alt="yuque"
                  />
                ),
                title: '云凤蝶',
                url: 'https://yunfengdie.com',
                description: '中台建站平台',
                openExternal: true,
              },
            ],
          },
        ]}
        bottom="Made with ❤️ by AFX"
      />,
    );
    expect(wrapper.render()).toMatchSnapshot();
  });
});
