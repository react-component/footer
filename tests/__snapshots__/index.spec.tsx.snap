// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rc-footer render Footer with columnLayout and backgroundColor 1`] = `
LoadedCheerio {
  "0": <footer
    class="rc-footer rc-footer-dark"
    style="color: red; background-color: transparent;"
  >
    <section
      class="rc-footer-container"
    >
      <section
        class="rc-footer-columns"
        style="justify-content: space-between;"
      >
        <div
          class="rc-footer-column"
        >
          <h2>
            title
          </h2>
        </div>
        <div
          class="rc-footer-column"
        >
          <h2>
            title
          </h2>
        </div>
      </section>
    </section>
  </footer>,
  "_root": LoadedCheerio {
    "0": Document {
      "children": [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`rc-footer render empty Footer 1`] = `
LoadedCheerio {
  "0": <footer
    class="rc-footer rc-footer-dark"
  >
    <section
      class="rc-footer-container"
    />
  </footer>,
  "_root": LoadedCheerio {
    "0": Document {
      "children": [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`rc-footer render light theme Footer 1`] = `
LoadedCheerio {
  "0": <footer
    class="rc-footer rc-footer-light"
    style="color: red;"
  >
    <section
      class="rc-footer-container"
    >
      <section
        class="rc-footer-columns"
      >
        <div
          class="rc-footer-column"
        >
          <h2>
            title
          </h2>
        </div>
        <div
          class="rc-footer-column"
        >
          <h2>
            title
          </h2>
        </div>
      </section>
    </section>
  </footer>,
  "_root": LoadedCheerio {
    "0": Document {
      "children": [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
